docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+  The generator has to create even larger islands for this starting positions
+  mode. Depending on the landmass settings, these large islands take up a
+  different fraction of available land: 50% for a landmass above 60%, 60% for a
+  landmass above 40%, and 70% for a landmass below. This part of the landmass is
+  then distributed evenly among player islands, with two or three players per
+  island; but letting the islands shrink to 10% of their intended size if
+  needed to place them.
+
+  Smaller islands are then created in two different sizes, one of each per
+  player, while also letting those shrink to 10% of their size if needed.
+
+The generators become significantly slower with high ``landmass``, and simply
+refuse to proceed when it is too high (above 80 or 85%). In such cases, they
+fall back to ``RANDOM``, ``FRACTAL``, or ``ISLANDS`` with the ``SINGLE`` player
+placement strategy.
+
+Island Creation
+^^^^^^^^^^^^^^^
+
+To create an island, the generator first builds up its shape and then tries to
+place it on the map. The shape is created by starting from a single tile and
+randomly expanding the island by adding cardinally adjacent tiles until the
+desired size is reached. In addition to the random expansion, all tiles with
+four neighbors are automatically added to the island; this prevents gaps from
+appearing in the middle.
+
+With the island shape created, the algorithm then tries to place it on the map
+by randomly selecting a tile as center and checking for collisions with the map
+boundaries and other islands. This is tried many times before the algorithm
+eventually gives up. Once a fitting location is found, the tiles at the map
+location are set to "unknown" terrain (the map is initially only water), then
+its terrain is immediately set.
+
+Rivers and Terrain
+^^^^^^^^^^^^^^^^^^
+
+Once an island is placed, rivers are added to it. The number of river tiles is
+computed worldwide and each island gets a fraction of it proportional to its
+surface. Creating rivers is done by repeatedly picking a random tile in the
+island. The first suitable river mouth found in this way becomes a river;
+suitable river mouths are coastal tiles with a single cardinally adjacent
+oceanic tile, at most 3 ocean tiles around, and no adjacent river. Afterwards,
+new river mouths are only added with a 1% probability. Instead, when the random
+tile is next to an existing river, the river grows in this direction. Further
+conditions for rivers to grow require that the new tile has no cardinally
+adjacent ocean and at most one adjacent ocean, and furthermore at most two
+adjacent rivers. Dry tiles are also penalized by dropping them with a 50%
+chance. In this way the rivers slowly grow inwards.
+
+Terrains are placed in a similar way: tiles are picked at random and terrain is
+assigned to them until reaching their expected number. This is done by groups of
+terrains, starting with forest-like, then desert-like, mountain-like, and
```

Comment:
s/terrains/terrain

Reply:
This is also about terrain "types" from the ruleset